packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -5824,12 +5824,15 @@ class ParentDataElement<T extends ParentData> extends ProxyElement {
   /// Creates an element that uses the given widget as its configuration.
   ParentDataElement(ParentDataWidget<T> super.widget);
 
+  /// Returns tne [Type] of [ParentData] that this element has been configured
+  /// for.
+  Type get parentDataType => T;
```

Comment:
should this be debugParentDataType and only return it in debug mode? I presume we only use this for a debug check? It should likely not be used for anything in production?